haystack/components/generators/chat/hugging_face_local.py

```diff
@@ -322,17 +363,23 @@ def create_message(  # pylint: disable=too-many-positional-arguments
         :param tokenizer: The tokenizer used for generation.
         :param prompt: The prompt used for generation.
         :param generation_kwargs: The generation parameters.
+        :param parse_tool_calls: Whether to attempt parsing tool calls from the text.
         :returns: A ChatMessage instance.
         """
+
         completion_tokens = len(tokenizer.encode(text, add_special_tokens=False))
         prompt_token_count = len(tokenizer.encode(prompt, add_special_tokens=False))
         total_tokens = prompt_token_count + completion_tokens
 
-        # not the most sophisticated finish_reason detection, improve later to match
-        # https://platform.openai.com/docs/guides/text-generation/chat-completions-response-format
-        finish_reason = (
-            "length" if completion_tokens >= generation_kwargs.get("max_new_tokens", sys.maxsize) else "stop"
-        )
+        tool_calls = self._parse_tool_call(text) if parse_tool_calls else None
```

Comment:
the problem I can see also running the notebook is that the final message will contain both `text` and parsed `tool_calls`.

`{'replies': [ChatMessage(_role=<ChatRole.ASSISTANT: 'assistant'>, _content=[TextContent(text='<tool_call>\n{"name": "weather", "arguments": {"city": "Paris"}}\n</tool_call>'), ToolCall(tool_name='weather', arguments={'city': 'Paris'}, id=None)], _name=None, ...}]]}`

What do you think about removing `text` if `tool_calls` are detected in the textual content?

Reply:
Deal 🙏